src/ReasonReactRouter.re

```diff
@@ -133,14 +144,42 @@ type url = {
   search: string,
 };
 type watcherID = unit => unit;
-let url = () => {path: path(), hash: hash(), search: search()};
+let urlFromWindow = () => {
+  path: pathFromWindow(),
+  hash: hashFromWindow(),
+  search: searchFromWindow(),
+};
+let url = (~pathname=?, ~search=?, ~hash=?, ()) => {
+  path:
+    pathFromString(
+      switch (pathname) {
+      | Some(pathname) => pathname
+      | None => "/"
+      },
+    ),
+  search:
+    searchFromString(
+      switch (search) {
+      | Some(search) => search
+      | None => ""
+      },
+    ),
+  hash:
+    hashFromString(
```

Comment:
This is an odd choice to me - [URL](https://nodejs.org/api/url.html#url_url_hash) in JS provides a hash without the leading `#` - so using this requires some extra effort. This is especially weird for server use since we don't have this value on the server so you'd have to be constructing from some other information. Seems like bad form. But I get why this PR would have it for completeness.